src/app/Controllers/ParameterValidator.cs

```diff
@@ -179,10 +282,101 @@ public static JsonResult MovieId(string movieId, string method, ILogger logger)
                 !int.TryParse(movieId.Substring(2), out int val) ||
                 val <= 0)
             {
-                return GetAndLogBadParam("Invalid Movie ID parameter", method, logger);
+                return GetAndLogInvalidMovieIdParameter(method, logger);
             }
 
             return null;
         }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="statusCode"></param>
+        /// <param name="errorCode"></param>
+        /// <param name="errorMessage"></param>
+        /// <param name="target"></param>
+        /// <param name="innerErrorType"></param>
+        /// <param name="minValue"></param>
+        /// <param name="maxValue"></param>
+        /// <returns></returns>
+        private static IActionResult BuildJsonResponse(HttpStatusCode statusCode, string errorCode, string errorMessage, string target,
+            InnerErrorType innerErrorType, object minValue, object maxValue)
+        {
+            var sb = new StringBuilder();
+            sb.Append("{\n");
+            sb.Append("\"error\": {\n");
+            sb.Append($"\"code\": \"{errorCode}\",\n");
+            sb.Append($"\"message\": \"{errorMessage}\",\n");
+            sb.Append($"\"statusCode\": {(int)statusCode},\n");
+            sb.Append($"\"target\": \"{target}\",\n");
+            sb.Append("\"innererror\": {\n");
+            sb.Append($"\"code\": \"{innerErrorType}\",\n");
+
+            switch (innerErrorType)
+            {
+                case InnerErrorType.InvalidSearchParameter:
+                    sb.Append($"\"minLength\": {minValue},\n");
```

Comment:
Why use string builder vs. just a string.format?  The format template would be human readable

Reply:
@bartr I followed your request from the previous comment.